src/components/structures/MessagePanel.js

```diff
@@ -430,11 +430,11 @@ module.exports = React.createClass({
         // as 'today' for the date separators.
```

Comment:
I'm not sure I see an issue. The `ts1` given to the `DateSeperator` etc. is a number of millis, not a date instance.